### PR TITLE
feat: support glob

### PR DESCRIPTION
**Pull request by JustroX**
This PR adds support for specifying custom extensions for migration scripts. This is somehow related to #72, as we have a similar use case, i.e. for testing migrations. However, I only implemented filtering by file extensions/file name suffixes instead of glob.

Also, if the file extension is specified in the config, the generated migration script's filename will also have that extension.

Your reviews and insights are highly appreciated. Thank you!

### CODE REVIEW

**Comment by mycodeself** [comment]
Hi Justrox, 

thanks a lot for your contribution, seems some tests are failing... anyway I think using a glob pattern will be a better implementation for this, what do you think?

See the discussion #72

**Comment by JustroX** [comment]
Thanks. Apologies for the late response, been buried in my work. I agree that glob patterns are better, I just initially implemented file name suffixes as it's enough for my use case. I'm still up for coding glob patterns tho :) I'll try tonight

**Comment by JustroX** [comment]
Implemented. Let me know what you think :)

**Comment by GerardSopsop** [comment]
Hello, my current team uses MongoDB and we badly need this feature. Any updates on this? Thanks' a lot man, @mycodeself

File: lib/config.ts [hunk 1]
```diff
@@ -29,6 +33,9 @@ export interface Config {
     databaseVar?: string;
   };
   options?: MongoClientOptions;
+
```
Comment (mycodeself) [inline lib/config.ts]:
```suggestion
```

remove this blank line

File: lib/config.ts [hunk 2]
```diff
@@ -16,6 +17,9 @@ export interface ProcessedConfig {
   migrationsDir: string;
   migrationsCollection: string;
   options?: MongoClientOptions;
+
```
Comment (mycodeself) [inline lib/config.ts]:
```suggestion
```
remove this blank line

File: lib/commands/up.ts [hunk 3]
```diff
@@ -28,7 +34,7 @@ export const up = async (opts: CommandUpOptions): Promise<void> => {
   try {
     const collection = connection.getMigrationsCollection(migrationsCollection);
     const appliedMigrations = await getAppliedMigrations(collection);
-    const migrations = (await loadMigrations(migrationsDir)).filter(
+    const migrations = (await loadMigrations(migrationsDir, pattern)).filter(
```
Comment (mycodeself) [inline lib/commands/up.ts]: is there the `glob` argument from the config missing?

**Review by mycodeself — CHANGES_REQUESTED** [review]
Hi @JustroX 

thanks a lot for your contribution. I suggested few changes, I think you're missing to update the `down` command and I'm wondering if would be needed to update also the `new` command.

Could you take a look?

**Comment by JustroX** [comment]
I'll take a look thanks

**Comment by mycodeself** [comment]
Hello @JustroX, I've opened a new PR including support for glob #95 but I'd like your contribution to be taken into account, so I leave you a few days in case you want to see my proposal and apply the suggested changes.

**Comment by JustroX** [comment]
Sorry for the delay, work has been keeping me busy. I haven't had a chance
to start on it yet, so please feel free to proceed with the pull request.

Great work :) Thanks a bunch!

File: lib/commands/up.ts (hunk 3 above)
Comment (JustroX) [reply lib/commands/up.ts]: yes, it's missing. my bad

**Comment by JustroX** [comment]
@mycodeself 
If ever, may I ask what changes are needed for the `down` and `new` commands? For the `down` command, I think the file reference was saved in the database. For the `new` command, I don't think we can unambiguously know where should the file be placed solely from the glob pattern.

**Comment by mycodeself** [comment]
Hey, sorry for the delay I was really busy, but I implemented the changes needed to have this working. Thanks for your contribution.

**Pull request closed without merging**